plugins/woocommerce/includes/bis/class-wc-bis-sync-tasks.php

```diff
@@ -0,0 +1,379 @@
+<?php
+/**
+ * WC_BIS_Sync_Tasks class
+ *
+ * @package  WooCommerce Back In Stock Notifications
+ * @since    1.0.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Sync stock Queue Controller Class.
+ *
+ * @class    WC_BIS_Sync_Tasks
+ * @version  x.x.x
+ */
+class WC_BIS_Sync_Tasks {
+
+	/**
+	 * Setup.
+	 */
+	public static function init() {
+
+		// Main triggers for stock tasks.
+		add_action( 'woocommerce_bis_sync_handle_outofstock_products', array( __CLASS__, 'handle_outofstock_products' ) );
+		add_action( 'woocommerce_bis_sync_handle_instock_products', array( __CLASS__, 'handle_instock_products' ) );
+
+		// Async tasks.
+		add_action( 'wc_bis_daily', array( __CLASS__, 'do_wc_bis_daily' ) );
+		add_action( 'wc_bis_process_notifications_batch', array( __CLASS__, 'async_process_notifications_batch' ), 10, 2 );
+
+		// Sync product delete.
+		add_action( 'before_delete_post', array( __CLASS__, 'handle_product_delete' ), 0 );
+
+		// Force queue notifications.
+		add_action( 'admin_init', array( __CLASS__, 'force_queue_notifications' ) );
+	}
+
+	/*****************************************
+	 * Instock tasks.
+	 ****************************************/
+
+	/**
+	 * Prepare notifications, mark queue status.
+	 *
+	 * @param  array $product_ids
+	 * @param  bool  $force (Optional)
+	 * @return void
+	 */
+	public static function handle_instock_products( $product_ids, $force = false ) {
+		if ( ! is_array( $product_ids ) ) {
+			$product_ids = array( $product_ids );
+		}
+
+		$last_sent_throttle            = wc_bis_get_minimum_time_between_notifications();
+		$throttle_timestamp            = $force || 0 === $last_sent_throttle ? 0 : time() - $last_sent_throttle;
+		$count_results                 = WC_BIS()->db->notifications->get_subscribed_notifications_count_per_product( $product_ids, $throttle_timestamp );
+		$notifications_count           = $count_results['allowed'];
+		$throttled_notifications_count = $count_results['throttled'];
+
+		// Check for recently sent notifications.
+		if ( $throttled_notifications_count > 0 && class_exists( 'WC_BIS_Admin_Notices' ) ) {
+			self::handle_spam_notices( $product_ids, $throttled_notifications_count );
+		}
+
+		if ( 0 === $notifications_count ) {
+			return;
+		}
+
+		if ( wc_bis_debug_enabled() ) {
+			WC_BIS()->log( sprintf( 'Sending %d notifications for products %s ', $notifications_count, implode( ',', $product_ids ) ), 'info', 'wc_bis_sync_logs' );
```

Comment:
It would be nice to just use `wc_get_logger()` directly here, and also clean this up to use [logging best practices](https://developer.woocommerce.com/docs/logging-in-woocommerce/#3-logging-best-practices).